src/components/producerItems/components/ProductList.js

```diff
@@ -43,7 +43,23 @@ export default class ProductList extends React.Component {
                 renderItem={item => {
                     const { description, _id, imageUrl, name } = item;
                     const actions = [
-                        <Icon key={EDIT} onClick={() => openModal(_id)} type={EDIT} />,
+                        <Icon
+                            key={EDIT}
+                            onClick={() => openModal(_id)}
+                            type={EDIT}
+                        />,
+                        <Popconfirm
+                            key={DELETE}
+                            title="Are you sure delete this product?"
+                            onConfirm={() => onConfirm(_id)}
+                            onCancel={() => onCancel()}
```

Comment:
`onCancel={onCancel}`

Reply:
ok, but i did that so as to prevent the action from firing

Reply:
you are passing the `onCancel` function not firing it. the component is handle firing it